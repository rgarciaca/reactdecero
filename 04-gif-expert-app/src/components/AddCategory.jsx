import { useState } from "react"
import PropTypes from "prop-types";

export const AddCategory = ( { onNewCategory } ) => {

    const [inputValue, setInputValue] = useState('');

    const onInputValue = ( event ) => {
        setInputValue( event.target.value );
    }

    const onSubmit = ( event ) => {
        event.preventDefault();

        if( inputValue.trim().length <= 1 ) return;

        onNewCategory( inputValue.trim() );
        setInputValue('');
    }

  return (
    <form onSubmit={ onSubmit }>
        <input type="text" plaholder="Buscar gifs" value={ inputValue } onChange={ onInputValue }></input>
    </form>
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};

