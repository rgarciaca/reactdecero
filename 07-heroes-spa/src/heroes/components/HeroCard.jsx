import { Link } from "react-router-dom";

export const HeroCard = ( { hero } ) => {

  const heroImageUrl = `/assets/heroes/${ hero.id}.jpg`;

  return (
        <div className="col  animate__animated animate__fadeIn">
          <div className="card">
            <div className="row no-gutters">
              <div className="col-4">
                <img src={ heroImageUrl } className="card-img" alt={ hero.superhero}></img>
              </div>
              <div className="col-8">
                <div className="card-body">
                  <h5 className="card-title">{ hero.superhero }</h5>
                  <p className="card-text">{ hero.alter_ego }</p>

                  { hero.alter_ego !== hero.characters && <p>{ hero.characters }</p> }

                  <p className="card-text">
                    <small className="text-muted">{ hero.first_appearance }</small>
                  </p>

                  <Link to={ `/hero/${ hero.id }`} >Mas...</Link>
                </div>
              </div>
            </div>
          </div>
        </div>        
  )
}
