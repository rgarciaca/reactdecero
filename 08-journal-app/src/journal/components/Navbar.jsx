import { LogoutOutlined, MenuOutlined } from "@mui/icons-material"
import { AppBar, Grid, IconButton, Toolbar, Typography } from "@mui/material"
import { useDispatch } from "react-redux"
import { startLogout } from "../../store/auth";


export const Navbar = ({ drawerWidth = 240 }) => {

    const dispatch = useDispatch();

    const onLogout = () => {
        dispatch( startLogout() );
    }

  return (
    <AppBar position='fixed' sx={{ width: { sm: `calc(100% - ${ drawerWidth }px)` }, ml: { drawerWidth } }}>
        <Toolbar>
            <IconButton color='inherit' edge='start' sx={{ mr: 2, display: { sm: 'none' } }} >
                <MenuOutlined sx={{ color: 'white' }}></MenuOutlined>
            </IconButton>

            <Grid container direction='row' justifyContent='space-between' alignItems='center'>
                <Typography variant='h6' nowrap='true' component='div'>JournalApp</Typography>

                <IconButton color='error' onClick={ onLogout }>
                    <LogoutOutlined></LogoutOutlined>
                </IconButton>
            </Grid>
        </Toolbar>
    </AppBar>
  )
}
