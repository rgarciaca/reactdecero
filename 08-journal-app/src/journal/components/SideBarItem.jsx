import { TurnedInNot } from "@mui/icons-material"
import { Grid, ListItem, ListItemButton, ListItemIcon, ListItemText } from "@mui/material"
import { useDispatch } from "react-redux";
import { setActiveNote } from "../../store/journal/JournalSlice";

export const SideBarItem = ( { note } ) => {

  const dispatch = useDispatch();

  const onClickNote = () => {
    dispatch( setActiveNote( note ));
  }


  return (
    <ListItem disablePadding >
        <ListItemButton onClick={ onClickNote }>
            <ListItemIcon>
                <TurnedInNot />
            </ListItemIcon>
            <Grid container>
                <ListItemText primary={ note.title } />
                <ListItemText secondary={ note.body }></ListItemText>
            </Grid>
        </ListItemButton>
    </ListItem>
  )
}
