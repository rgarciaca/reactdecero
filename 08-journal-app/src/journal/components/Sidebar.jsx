import { LocalFireDepartment, TurnedInNot } from "@mui/icons-material"
import { Box, Divider, Drawer, Grid, List, ListItem, ListItemButton, ListItemIcon, ListItemText, Toolbar, Typography } from "@mui/material"
import { useSelector } from "react-redux"

export const Sidebar = ({ drawerWidth = 240 }) => {

    const { displayName } = useSelector( state => state.auth );

  return (
    <Box component='nav' sx={{ width: { sm: drawerWidth }, flexShrink: { sm: 0 } }}>
        <Drawer variant='permanent' open sx={{ displsy: { xs: 'block' }, '& .MuiDrawer-paper': { boxSizing: 'border-box', width: drawerWidth } }}>
            <Toolbar>
                <Typography variant='h6' nowrap="true" component='div' sx={{ fontSize: '1.0rem' }} >{ displayName }</Typography>
            </Toolbar>
            <Divider />

            <List>
                {
                    ['Enero', 'Febrero', 'Marzo', 'Abril'].map( text => (
                        <ListItem key={ text } disablePadding >
                            <ListItemButton>
                                <ListItemIcon>
                                    <TurnedInNot />
                                </ListItemIcon>
                                <Grid container>
                                    <ListItemText primary={ text } />
                                    <ListItemText secondary={ 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Magni dolor voluptatibus labore! ' }></ListItemText>
                                </Grid>
                            </ListItemButton>
                        </ListItem>
                    ))
                }
            </List>
        </Drawer>
    </Box>
  )
}
