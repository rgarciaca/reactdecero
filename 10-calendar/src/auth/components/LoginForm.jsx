import { useEffect } from "react";
import { useAuthStore, useForm } from "../../hooks"
import Swal from "sweetalert2";


const loginFormFields = {
    loginEmail:     '',
    loginPassword:  '',

}

export const LoginForm = () => {

    const { startLogin, errorMessage } = useAuthStore();

    const { loginEmail, loginPassword, onInputChange} = useForm( loginFormFields );

    const loginSubmit = ( event ) => {
        event.preventDefault();
        startLogin({ email: loginEmail, password: loginPassword });

    }

    useEffect(() => {
      if ( errorMessage !== undefined ) {
        Swal.fire('Error en la autenticación', errorMessage, 'error');
      }
    

    }, [errorMessage])
    

    return (
        <>
            <h3>Inicio de sesión<nav></nav></h3>
            <form onSubmit={ loginSubmit }> 
                <div className="form-group mb-2">
                    <input 
                        type="text"
                        className="form-control"
                        placeholder="Correo"
                        name="loginEmail"
                        value={ loginEmail }
                        onChange={ onInputChange}
                    />
                </div>
                <div className="form-group mb-2">
                    <input
                        type="password"
                        className="form-control"
                        placeholder="Contraseña"
                        name="loginPassword"
                        value={ loginPassword }
                        onChange={ onInputChange}
                    />
                </div>
                <div className="form-group mb-2">
                    <input 
                        type="submit"
                        className="btnSubmit"

                        value="Login" 
                    />
                </div>
            </form>    
        </>
    )
}
