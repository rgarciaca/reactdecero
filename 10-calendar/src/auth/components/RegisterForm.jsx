import Swal from "sweetalert2";
import { useAuthStore, useForm } from "../../hooks";
import { useEffect } from "react";


const registerFormFields = {
    registerName:       '',       
    registerEmail:      '',
    registerPassword:   '',
    registerPassword2:  '',
}


export const RegisterForm = () => {

    const { startRegister, errorMessage } = useAuthStore();
    const { registerName, registerEmail, registerPassword, registerPassword2, onInputChange } = useForm( registerFormFields );

    const registerSubmit = ( event ) => {
        event.preventDefault();
        if ( registerPassword !== registerPassword2 ) {
            Swal.fire('Error en registro de usuario', 'Las contraseñas no son iguales', 'error');
            return;
        }

        startRegister({ name: registerName, email: registerEmail, password: registerPassword });
    }

    useEffect(() => {
      if ( errorMessage !== undefined ) {
        Swal.fire('Error en el registro de usuario', errorMessage, 'error');
      }
    

    }, [errorMessage])

    return (
        <>
            <h3>Registro</h3>
            <form onSubmit={ registerSubmit }>
                <div className="form-group mb-2">
                    <input
                        type="text"
                        className="form-control"
                        placeholder="Nombre"
                        name="registerName"
                        value={ registerName }
                        onChange={ onInputChange}                               
                    />
                </div>
                <div className="form-group mb-2">
                    <input
                        type="email"
                        className="form-control"
                        placeholder="Correo"
                        name="registerEmail"
                        value={ registerEmail }
                        onChange={ onInputChange}                        
                    />
                </div>
                <div className="form-group mb-2">
                    <input
                        type="password"
                        className="form-control"
                        placeholder="Contraseña" 
                        name="registerPassword"
                        value={ registerPassword }
                        onChange={ onInputChange}
                    />
                </div>

                <div className="form-group mb-2">
                    <input
                        type="password"
                        className="form-control"
                        placeholder="Repita la contraseña" 
                        name="registerPassword2"
                        value={ registerPassword2 }
                        onChange={ onInputChange}                               
                    />
                </div>

                <div className="form-group mb-2">
                    <input 
                        type="submit" 
                        className="btnSubmit" 
                        value="Crear cuenta" />
                </div>
            </form>    
        </>
    )
}
