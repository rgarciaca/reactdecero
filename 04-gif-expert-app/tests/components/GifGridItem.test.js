
import { render, screen } from '@testing-library/react';
import { GifGridItem } from '../../src/components/GifGridItem';


describe('Pruebas en <GifGridItem />', () => {

    const title = 'Smeagol';
    const url = 'https://elmundo.es/';

    test("debe de hacer match con el snapshot", () => {       

        const { container} = render( <GifGridItem title={ title } url={ url }/>);
        expect( container ).toMatchSnapshot();
    });

    test("debe de mostrar la imagen con el url y el alt indicado", () => {       

        render( <GifGridItem title={ title } url={ url }/>);
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe(title);
    });

    test("debe de mostrar el titulo en el componente", () => {       

        render( <GifGridItem title={ title } url={ url }/>);
        expect( screen.getByText( title ) ).toBeTruthy();
    });
});