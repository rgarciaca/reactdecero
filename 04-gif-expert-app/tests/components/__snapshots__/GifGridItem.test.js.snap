// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Smeagol"
      src="https://elmundo.es/"
    />
    <p>
      Smeagol
    </p>
  </div>
</div>
`;
